libcudacxx/include/cuda/std/__cmath/logarithms.h

```diff
@@ -0,0 +1,494 @@
+#endif // !_CCCL_BUILTIN_LOG
+}
+
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI float logf(float __x) noexcept
+{
+#if defined(_CCCL_BUILTIN_LOG)
+  return _CCCL_BUILTIN_LOGF(__x);
+#else // ^^^ _CCCL_BUILTIN_LOG ^^^ / vvv !_CCCL_BUILTIN_LOG vvv
+  return ::logf(__x);
+#endif // !_CCCL_BUILTIN_LOG
+}
+
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI double log(double __x) noexcept
+{
+#if defined(_CCCL_BUILTIN_LOG)
+  return _CCCL_BUILTIN_LOG(__x);
+#else // ^^^ _CCCL_BUILTIN_LOG ^^^ / vvv !_CCCL_BUILTIN_LOG vvv
+  return ::log(__x);
+#endif // !_CCCL_BUILTIN_LOG
+}
+
+#if !defined(_LIBCUDACXX_HAS_NO_LONG_DOUBLE)
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI long double log(long double __x) noexcept
+{
+#  if defined(_CCCL_BUILTIN_LOG)
+  return _CCCL_BUILTIN_LOGL(__x);
+#  else // ^^^ _CCCL_BUILTIN_LOG ^^^ / vvv !_CCCL_BUILTIN_LOG vvv
+  return ::logl(__x);
+#  endif // !_CCCL_BUILTIN_LOG
+}
+
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI long double logl(long double __x) noexcept
+{
+#  if defined(_CCCL_BUILTIN_LOG)
+  return _CCCL_BUILTIN_LOGL(__x);
+#  else // ^^^ _CCCL_BUILTIN_LOG ^^^ / vvv !_CCCL_BUILTIN_LOG vvv
+  return ::logl(__x);
+#  endif // !_CCCL_BUILTIN_LOG
+}
+#endif // !_LIBCUDACXX_HAS_NO_LONG_DOUBLE
+
+#if defined(_LIBCUDACXX_HAS_NVFP16)
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI __half log(__half __x) noexcept
+{
+  NV_IF_ELSE_TARGET(NV_PROVIDES_SM_53, (return ::hlog(__x);), ({
+                      float __vf            = __half2float(__x);
+                      __vf                  = _CUDA_VSTD::logf(__vf);
+                      __half_raw __ret_repr = ::__float2half_rn(__vf);
+
+                      _CUDA_VSTD::uint16_t __repr = __half_raw(__x).x;
+                      switch (__repr)
+                      {
+                        case 7544:
+                          __ret_repr.x -= 1;
+                          break;
+
+                        default:;
+                      }
+
+                      return __ret_repr;
+                    }))
+}
+#endif // _LIBCUDACXX_HAS_NVFP16
+
+#if defined(_LIBCUDACXX_HAS_NVBF16)
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI __nv_bfloat16 log(__nv_bfloat16 __x) noexcept
+{
+  NV_IF_ELSE_TARGET(
+    NV_IS_DEVICE, (return ::hlog(__x);), (return __float2bfloat16(_CUDA_VSTD::logf(__bfloat162float(__x)));))
+}
+#endif // _LIBCUDACXX_HAS_NVBF16
+
+template <class _Integer, enable_if_t<_CCCL_TRAIT(is_integral, _Integer), int> = 0>
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI double log(_Integer __x) noexcept
+{
+#if defined(_CCCL_BUILTIN_LOG)
+  return _CCCL_BUILTIN_LOG((double) __x);
+#else // ^^^ _CCCL_BUILTIN_LOG ^^^ / vvv !_CCCL_BUILTIN_LOG vvv
+  return ::log((double) __x);
+#endif // !_CCCL_BUILTIN_LOG
+}
+
+// log10
+
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI float log10(float __x) noexcept
+{
+#if defined(_CCCL_BUILTIN_LOG10)
+  return _CCCL_BUILTIN_LOG10F(__x);
+#else // ^^^ _CCCL_BUILTIN_LOG10 ^^^ / vvv !_CCCL_BUILTIN_LOG10 vvv
+  return ::log10f(__x);
+#endif // !_CCCL_BUILTIN_LOG10
```

Comment:
Suggestion: Even though you define/undef `_CCCL_BUILTIN_LOG10[|F|L]` in union, I 
```suggestion
#if defined(_CCCL_BUILTIN_LOG10F)
  return _CCCL_BUILTIN_LOG10F(__x);
#else // ^^^ _CCCL_BUILTIN_LOG10F ^^^ / vvv !_CCCL_BUILTIN_LOG10F vvv
  return ::log10f(__x);
#endif // !_CCCL_BUILTIN_LOG10F
```
This applies to all the log functions below.